astropy/io/fits/card.py

```diff
@@ -283,8 +282,7 @@ def value(self):
         if self._value is not None:
             value = self._value
         elif self._valuestring is not None or self._image:
-            self._value = self._parse_value()
-            value = self._value
+            value = self._value = self._parse_value()
```

Comment:
I would prefer the old style here.

Reply:
Why, multiple assignment is cool :p

Reply:
I agree with @saimn!

Reply:
:scream:

Reply:
Maybe it's just me, but it feels like an unintended misspelling of `value = self._value == self._parse_value()`. Also it reminds me of C.

Don't get me wrong I won't start a fight about it but this looks so weird to me because it's so rare that I see this in Python and since assignment is normally a statement not an expression it looks like it really shouldn't work.

Reply:
I don't know, I frequently use this, typically to get a shorter variable name for a local block and avoid the lookup time on self. It's quite practical, and imo easier to understand that it references the same variable.

Reply:
Okay, so it's probably just a matter of getting used to it. I think we probably have to get used to it when [Python 3.8 with PEP 572 -- Assignment Expressions](https://www.python.org/dev/peps/pep-0572/) comes.

Reply:
wow, I should read more PEP, this looks quite interesting (though I agree that feels very unusual)

Reply:
Ah yes, PEP 572, the one that led to Guido stepping down from BDFL! I thought it was strange at first but I constantly find myself needing it.